lib/chef/knife/azurerm_server_list.rb

```diff
@@ -11,7 +11,11 @@ class AzurermServerList < Knife
       def run
         $stdout.sync = true
         validate_arm_keys!
-        service.list_servers
+        if locate_config_value(:azure_resource_group_name).nil?
+          service.list_servers
+        else
+          service.list_servers(locate_config_value(:azure_resource_group_name))
+        end

```

Comment:
Seems like we can remove this `if` block since `list_servers` method will take care of handling the `nil` value.
Please correct me if I miss something.